app/controllers/reservations_controller.rb

```diff
@@ -21,20 +20,20 @@ def edit
 
   def create
     authorize Reservation
+
     @reservation = Reservation.new(screening_id: params[:screening_id], user_id: current_user.id, status: :created)
 
-    if !params.key?(:seats)
-      @reservation.errors.add(:base, 'You have to choose at least one seat')
-      render :new, status: :unprocessable_entity
-    else
-      @reservation.save
-      create_tickets
+    if CreateReservation.new(@reservation, params).call
```

Comment:
I think it would be better to create a `Reservation` object inside `CreateReservation`. This controller doesn't need to know how to create reservation on DB level.